mission/mission_planner.cpp

```diff
@@ -0,0 +1,1038 @@
+
+        mavlink_message_t msg;
+        mavlink_msg_mission_current_pack(mission_planner->mavlink_stream_.sysid(),
+                                         mission_planner->mavlink_stream_.compid(),
+                                         &msg,
+                                         mission_planner->waypoint_handler_.current_waypoint_index());
+        mission_planner->mavlink_stream_.send(&msg);
+
+        result = MAV_RESULT_ACCEPTED;
+    }
+    else
+    {
+        result = MAV_RESULT_TEMPORARILY_REJECTED;
+
+        print_util_dbg_print("Not ready to switch to next waypoint. Either no waypoint loaded or flying towards one\r\n");
+    }
+
+    // To avoid a MAV_RESULT_TEMPORARILY_REJECTED for the second message and thus
+    //  a bad information to the user on the ground, if two messages are received
+    //  in a short time interval, we still show the result as MAV_RESULT_ACCEPTED
+    if (time_from_start_wpt < time_wpt_limit)
+    {
+        result = MAV_RESULT_ACCEPTED;
+    }
+
+    return result;
+}
+
+void Mission_planner::set_current_waypoint(Mission_planner* mission_planner, uint32_t sysid, mavlink_message_t* msg)
+{
+    mavlink_mission_set_current_t packet;
+
+    mavlink_msg_mission_set_current_decode(msg, &packet);
+
+    // Check if this message is for this system and subsystem
+    if (((uint8_t)packet.target_system == (uint8_t)sysid)
+            && ((uint8_t)packet.target_component == (uint8_t)MAV_COMP_ID_MISSIONPLANNER))
+    {
+        if (mission_planner->waypoint_handler_.set_current_waypoint_index(packet.seq))
+        {
+            mavlink_message_t _msg;
+            mavlink_msg_mission_current_pack(sysid,
+                                             mission_planner->mavlink_stream_.compid(),
+                                             &_msg,
+                                             packet.seq);
+            mission_planner->mavlink_stream_.send(&_msg);
+
+            mission_planner->navigation_.set_start_wpt_time();
+            mission_planner->navigation_.set_waiting_at_waypoint(false);
+            mission_planner->switch_mission_handler(mission_planner->waypoint_handler_.current_waypoint());
+        }
+        else
+        {
+            mavlink_message_t _msg;
+            mavlink_msg_mission_ack_pack(mission_planner->mavlink_stream_.sysid(),
+                                         mission_planner->mavlink_stream_.compid(),
+                                         &_msg,
+                                         msg->sysid,
+                                         msg->compid,
+                                         MAV_CMD_ACK_ERR_ACCESS_DENIED);
+            mission_planner->mavlink_stream_.send(&_msg);
+        }
+    } //end of if this message is for this system and subsystem
+}
+
+mav_result_t Mission_planner::set_current_waypoint_from_parameter(Mission_planner* mission_planner, mavlink_command_long_t* packet)
```

Comment:
also renamed in https://github.com/lis-epfl/MAVRIC_Library/pull/331/commits/8e2545a2691d3196908240b5f42d1deb57781535